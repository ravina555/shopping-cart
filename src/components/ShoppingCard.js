import React,{useState, useEffect} from 'react';
import '../App.css';
import { Card, CardHeader, CardContent, CardMedia, Typography } from '@mui/material';
import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
import RemoveCircleOutlineIcon from '@mui/icons-material/RemoveCircleOutline';
import { CardActions, TextField,Button } from '@mui/material';
import SummaryCart from './CartSummary';


export default function ShoppingCard(props) { 
  let allProducts = props?.Products;

  const[ProductDetails , updateProduct] = useState([]);

  useEffect(()=>{
    let Products = allProducts.map((element)=>{
      return {...element , quantity:0}
  });
  updateProduct(Products);
  },[allProducts])

  const handleAddOrRemove = (e,action) =>{
    let currentItem = +(e.currentTarget.parentElement.parentElement.id) ;
    let newUpdatedCart = alterCart(currentItem,action);
    console.log(newUpdatedCart);
      updateProduct(newUpdatedCart)
  }

  const alterCart = (currentItem,action) =>{
    return ProductDetails.map((element)=>{
      let updateQuantity = action? element.quantity+1 : element.quantity-1;
      if(element.id === currentItem  && element.quantity>=0){
        return {...element , quantity:updateQuantity}
      }else{
        return {...element}
      }      
    });
  }

  return (
    <>
      {ProductDetails.length > 0 && ProductDetails.map((item,index)=>{
      return (
       
        <Card className='shoppingCardsWrapper' key={item.id} id={item.id}>
        <CardHeader title={item.title.split(' ')[0]}>
        </CardHeader>
        <CardContent>
          <CardMedia
          component="img"
          height="194"
          image={item.image}
          alt={item.description}
        />
          <Typography>Rate: {item.price}</Typography>
        </CardContent>
        <CardActions id={item.id}>
                <Button onClick={(e)=>handleAddOrRemove(e,'add')}><AddCircleOutlineIcon/></Button>
                <TextField id="outlined-basic" label='no. of products' value={item.quantity} variant="outlined"/>
                <Button onClick={(e)=>handleAddOrRemove(e)}><RemoveCircleOutlineIcon/></Button>
          </CardActions>
        
      </Card>
    
      )
    })}

    <SummaryCart updatedCart ={ProductDetails}/>
    </>

  );
}