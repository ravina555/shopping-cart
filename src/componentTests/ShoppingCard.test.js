import ShoppingCard from '../components/ShoppingCard';
import Enzyme,{shallow,mount, ReactWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()})

describe('it should render shopping card',()=>{
    const Products=[{
        category: "men's clothing",
        description: "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
        id: 1,
        image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
        price: 109.95,
        rating: {rate: 3.9, count: 120},
        title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
    }]
       
    it('should match snapshot',()=>{
        const card = renderer.create(<ShoppingCard />);
        expect(card).toMatchSnapshot();
    });
})