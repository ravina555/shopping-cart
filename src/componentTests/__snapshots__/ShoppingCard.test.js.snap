// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render shopping card should match snapshot 1`] = `null`;
